src/prog_models/sim_result.py

```diff
@@ -149,14 +154,33 @@ def is_cached(self):
         return self.__data is not None
 
     def clear(self):
+        """
+        Clears the times, states, and data cache for a LazySimResult object
+        """
         self.times = []
         self.__data = None
         self.states = []
 
     def extend(self, other):
-        self.times.extend(deepcopy(other.times))  # lgtm [py/modification-of-default-value]
-        self.__data = None
-        self.states.extend(deepcopy(other.states))  # lgtm [py/modification-of-default-value]
+        """
+        Extend the LazySimResult with another LazySimResult object
+        Raise ValueError if SimResult is passed
+
+        Args:
+            other (LazySimResult)
+
+        """
+        if (isinstance(other, self.__class__)):
+            self.times.extend(deepcopy(other.times))  # lgtm [py/modification-of-default-value]
+            self.states.extend(deepcopy(other.states))  # lgtm [py/modification-of-default-value]
+            if self.__data is None or not other.is_cached():
```

Comment:
Now that you added this feature, lets add a test for it. If you do the following:

* Create 2 Lazy Sim Results without accessing data -> extend one from the other -> check extended lazysimresult.is_cached - Should be false
* Create 2 Lazy Sim Results access 1st one  -> extend one from the other -> check extended lazysimresult.is_cached - Should be false
* Create 2 Lazy Sim Results access 2nd one  -> extend one from the other -> check extended lazysimresult.is_cached - Should be false
* * Create 2 Lazy Sim Results access both one  -> extend one from the other -> check extended lazysimresult.is_cached - Should be true